### ISSUE TITLE
[BUG] cmake with parallel not producing parallel libs

### ISSUE THREAD

**Issue by fkramr** [HDFGroup/hdf5#2327]
**Describe the bug**
When using cmake and ENABLE_PARALLEL, produced libraries are not parallel.

Steps to reproduce:
```
cmake -DBUILD_SHARED_LIBS=ON -DBUILD_STATIC_LIBS=ON -DBUILD_TESTING=OFF -DHDF5_BUILD_EXAMPLES=OFF -DHDF5_BUILD_TOOLS=OFF -DHDF5_BUILD_UTILS=OFF -DHDF5_ENABLE_PARALLEL=ON ../hdf5_src
make -j
nm bin/libhdf5.a | grep H5Pset_fapl_mpio
```
The last command returns an empty string i.e. the parallel symbol is missing.

**Expected behavior**
The last line should return something similar to:
0000000000146f90 T H5Pset_fapl_mpio

**Platform (please complete the following information)**
 - HDF5 version 1.12.1 (but #833/#860 applied)
 - OpenSUSE 15.1
 - gcc 7
 - Build system cmake 3.21.3
 - Any configure options you specified: see above
 - openmpi 1.10.7

**Additional context**
When I move the H5pubconf.h outside the source tree beforehand, the parallel library is built as expected. Also a serial building produces a serial library. So this leads to the expected behaviour.
So it seems to be a problem of precedence of the includes. The H5pubconf.h in the Binary directory created by the cmake configure step seems to be ignored.
However, changing the order of the includes in the target_include_directories of src/CMakeLists.txt did not solve the issue. So may be I am missing something. Sorry, if this is a trivial mistake of my own...

**Comment by derobins** [comment]
Does this bug exist in develop?

**Comment by jwsblokland** [comment]
Maybe you did this already, but it may be worthwhile to check `libhdf5.settings` file. It contains the options and flags which have been used during the compilation. In the past, it helped me to discover that by accident I built the debug instead of the release version of the library.

**Comment by fkramr** [comment]
Ok. I checked that it works on develop, and that the build type is Release.

But while checking different releases I understood my problem roughly: The H5pubconf.h in my original src tree came from a plain ./configure that I ran some weeks back. This artefact got precedence over the H5pubconf generated by cmake in the out-of-source build folder. (I wasn't aware that the file was not from the original sources. I should have made a "make distclean". Deleting the file manually was a surprisingly valid solution.)
Such a bogus H5pubconf makes trouble also with 1_13_3. But since the support for ./configure seems to have been dropped in 1_14 and upwards, this should be very unlikely to happen to anyone in the future...
Thanks for your suggestions and time!
